hazelcast/src/test/java/com/hazelcast/query/impl/predicates/InPredicateTest.java

```diff
@@ -68,4 +70,15 @@ public void equal_NaN() {
         assertFalse(Double.NaN == -Float.NaN);
         assertFalse(Float.NaN == -Double.NaN);
     }
+
+    @Test
+    public void testEqualsAndHashCode() {
+        EqualsVerifier.forClass(InPredicate.class)
+            .suppress(Warning.NONFINAL_FIELDS, Warning.STRICT_INHERITANCE)
+            .withRedefinedSuperclass()
+            //.allFieldsShouldBeUsed() cannot be used
+            // as we should not use the "convertedInValues" field for equality
```

Comment:
`convertedInValues` should be marked as `transient`: it's not part of `InPredicate`'s state, it is just a convenience cache.